ceph/ceph.py

```diff
@@ -1097,6 +1097,58 @@ def get_nodes_in_location(self, location: str) -> list:
         """Return the list of nodes found in the location."""
         return [node for node in self.node_list if node.vm_node.location == location]
 
+    def get_node_ip_by_hostname(self, hostname):
+        """
+
+        Args:
+            hostname:
+
+        Returns:
+
+        """
+        cmd = "hostname -I | awk '{print $1}'"
+        node_list = [node for node in self.get_nodes() if node.hostname == hostname]
+        node = node_list[0] if len(node_list) > 0 else None
+        return node.exec_command(cmd=cmd)[0].replace("\n", "")
+
+    def get_node_ips_by_role(self, role):
+        """
+
+        Args:
+            role:
+
+        Returns:
+
+        """
+        cmd = "hostname -I | awk '{print $1}'"
+        ceph_object_list = self.get_ceph_objects(role=role)
+        node_ips = []
+        for obj in ceph_object_list:
+            node_ips.append(obj.exec_command(cmd=cmd)[0].replace("\n", ""))
+        return node_ips
+
+    def get_cluster_fsid(self, rhbuild, cluster_name=None, client=None):
+        """
+
+        Returns:
+
+        """
+        cmd = "ceph fsid"
+        pacific = True if (rhbuild and rhbuild.split(".")[0] >= "5") else False
+        if not client:
+            client = (
+                self.get_ceph_object("client")
+                if self.get_ceph_object("client")
+                else self.get_ceph_object("mon")
+            )
+        if cluster_name is not None:
+            cmd += f" --cluster {cluster_name}"
+        if pacific:
+            cmd = f"cephadm shell -- {cmd}"
```

Comment:
Do we install cephadm on all clients by default ?

Reply:
I guess we do. We have used similar commands before in different methods as well, so it should work.